java/vector/src/main/java/org/apache/arrow/vector/DecimalVector.java

```diff
@@ -489,6 +490,36 @@ public void setSafe(int index, int isSet, int start, ArrowBuf buffer) {
     set(index, isSet, start, buffer);
   }
 
+  @Override
+  public boolean equals(int index, ValueVector to, int toIndex) {
+    if (to == null) {
+      return false;
+    }
+    if (this.getMinorType() != to.getMinorType()) {
+      return false;
+    }
+
+    DecimalVector that = (DecimalVector) to;
+
+    if (this.scale != that.scale || this.precision != that.precision) {
```

Comment:
this is not specific to decimals. I checked the cpp impl, they also do this for timestamp/interval types. https://github.com/apache/arrow/blob/master/cpp/src/arrow/compare.cc#L694

I think the equivalent in java would be to compare the ArrowType (available in vector.field.fieldType.type).

Reply:
Thanks for pointing, looks reasonable to me, PR updated.